dpm/client/__init__.py

```diff
@@ -122,23 +122,28 @@ def publish(self, publisher=None):
         self.validate()
         token = self._ensure_auth()
 
-        # TODO: (?) echo('Uploading datapackage.json ... ', nl=False)
         response = self._apirequest(
-                method='PUT',
-                url='/api/package/%s/%s' % (self.username, self.datapackage.descriptor['name']),
-                json=self.datapackage.descriptor
-                )
+                method='POST',
+                url='/api/datastore/authorize',
+                json=self._prepare_files_authorize()
+            )
+
+        filedata = response.json().get('filedata')
+        if not filedata:
+            raise DpmException('server did not provide upload authorization for files')
+        for item in sorted(filedata.keys()):
```

Comment:
We want to upload in a specific order IMO:

- datapackage.json
- README
- Resources in the order they are in in the resources list in the Data Package

That is logical for the user and also means small files first.

Does this code respect that order?